docs/content/operator/gas-station/architecture/features.mdx

```diff
@@ -114,6 +114,232 @@ flowchart TD
+    - sender-address: "*"
+      gas-usage:
+        value: <1000000
+        window: 1 day
+        count-by: sender-address
+      action: allow
+```
+
+Here’s what happens:
+
+* The policy is set to `deny-all`, meaning only explicitly allowed transactions are processed.
+* The rule matches any sender address.
+* The `gas-usage` condition specifies a 1-day rolling window and limits usage to below 1,000,000 gas units per address.
+* The `count-by: sender-address` clause ensures that gas usage is tracked and limited separately for each sender.
+
+##### Why `count-by` is Important
+The `count-by` field is crucial for isolating gas usage limits per sender address. It ensures that each sender has their own independent gas usage counter, preventing one user's activity from affecting another's access to the gas station.
+This is particularly important in scenarios where multiple users might be interacting with the gas station simultaneously. If `count-by` were omitted, the gas usage would be aggregated across all senders, leading to potential abuse or unfair distribution of resources.
+This means that if one user consumes a significant portion of the gas limit, it does not impact other users' ability to access the gas station. Each sender's gas usage is tracked independently, allowing for fair and equitable access to resources.
+
+
+#### Combining Multiple Rules with Separate Counters
+
+You can define multiple rules with independent gas usage counters. Each rule is evaluated individually and maintains its own gas usage state, enabling precise control over different conditions.
+
+```yaml
+rules:
+  - sender-address: 0x0202020202020202020202020202020202020202020202020202020202020202
+    move-call-package-address: 0x0101010101010101010101010101010101010101010101010101010101010101
+    gas-usage:
+      value: <1500000
+      window: 1 day
+    action: allow
+
+  - sender-address: "*"
+    gas-usage:
+      value: <1000000
+      window: 1 day
+      count-by: sender-address
+    action: allow
+```
+
+Let’s walk through how the transaction processing works with this setup:
+
+Suppose an account with the address `0x0202...` initiates a transaction targeting the package `0x0101...`. If the transaction requires, for instance, 1,400,000 gas units, it will be accepted by the **first rule** because:
+
+* The sender address matches exactly.
+* The package address matches.
+* The gas requirement is below the 1,500,000 limit.
+
+```mermaid
+flowchart TD
+    A["Transaction"] --> B{"Match Rule 1?"}
+    B -- "Yes: sender + package + gas usage OK" --> C["Apply Rule 1: Allow"]
+
+    style A stroke-width:3px
+    style B stroke:#4CAF50,stroke-width:3px
+    style C stroke:#4CAF50,stroke-width:3px
+```
+
+Now, imagine the same account initiates another transaction, this time requiring 1,000,000 gas units. The **first rule** does not apply because the `gas-limit` matcher would be above the 1,500,000. The transaction then proceeds to the **second rule**, which:
```

Comment:
1,000,000 > 1,500,000?

```suggestion
Now, imagine the same account initiates another transaction, this time requiring 2,000,000 gas units. The **first rule** does not apply because the `gas-limit` matcher would be above the 1,500,000. The transaction then proceeds to the **second rule**, which:
```